python-frontend/typeshed_serializer/tests/test_symbols_merger.py

```diff
@@ -0,0 +1,195 @@
+            assert_merged_overloaded_functions_to_proto(merged_class_proto.overloaded_methods,
+                                                        original_class_symbol.overloaded_methods)
+        assert merged_class_proto.valid_for == original_class_symbol.valid_for
+
+
+def assert_merged_function_symbol_to_proto(merged_functions_proto, merged_functions):
+    assert len(merged_functions_proto) == len(merged_functions)
+    for merged_function_proto in merged_functions_proto:
+        matching_function_symbol = merged_functions[merged_function_proto.fully_qualified_name]
+        assert len(matching_function_symbol) == 1
+        original_function_symbol = matching_function_symbol[0]
+        assert merged_function_proto.name == original_function_symbol.function_symbol.name
+        assert merged_function_proto.fully_qualified_name == original_function_symbol.function_symbol.fullname
+        assert merged_function_proto.has_decorators == original_function_symbol.function_symbol.has_decorators
+        assert (merged_function_proto.resolved_decorator_names
+                == original_function_symbol.function_symbol.resolved_decorator_names)
+        assert merged_function_proto.is_abstract == original_function_symbol.function_symbol.is_abstract
+        assert merged_function_proto.is_asynchronous == original_function_symbol.function_symbol.is_asynchronous
+        assert merged_function_proto.is_final == original_function_symbol.function_symbol.is_final
+        assert merged_function_proto.is_overload == original_function_symbol.function_symbol.is_overload
+        assert merged_function_proto.is_property == original_function_symbol.function_symbol.is_property
+        assert merged_function_proto.is_static == original_function_symbol.function_symbol.is_static
+        assert merged_function_proto.is_class_method == original_function_symbol.function_symbol.is_class_method
+        assert merged_function_proto.valid_for == original_function_symbol.valid_for
+
+
+def assert_merged_overloaded_functions_to_proto(merged_overloaded_functions_proto, merged_overloaded_functions):
+    assert len(merged_overloaded_functions_proto) == len(merged_overloaded_functions)
+    for merged_overloaded_function_proto in merged_overloaded_functions_proto:
+        matching_overloaded_function = merged_overloaded_functions[merged_overloaded_function_proto.fullname]
+        assert len(matching_overloaded_function) == 1
+        original_overloaded_function = matching_overloaded_function[0]
+        assert merged_overloaded_function_proto.name == original_overloaded_function.overloaded_function_symbol.name
+        assert (merged_overloaded_function_proto.fullname
+               == original_overloaded_function.overloaded_function_symbol.fullname)
+        assert (len(merged_overloaded_function_proto.definitions)
+               == len(original_overloaded_function.overloaded_function_symbol.definitions))
```

Comment:
are those tests really useful? it looks like we're testing protobuf itself

Reply:
They're not the most interesting tests, to say the least. But they're testing that the `to_proto` methods are correctly mapping all the relevant field. Before writing those tests, I always seemed to forget some field in the mapping, so I'd say it's useful to have them 😄